pkgs/applications/editors/xournalpp/default.nix

```diff
@@ -0,0 +1,77 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+
+, cmake
+, gettext
+, makeWrapper
+, pkgconfig
+, git
+
+, glib
+, gsettings-desktop-schemas
+, gtk3
+, libsndfile
+, libxml2
+, pcre
+, poppler
+, portaudio
+, zlib
+
+# Plugins don't appear to be working in this version, so disable them by not
+# building with Lua support by default. In a future version, try switching this
+# to 'true' and seeing if the top-level Plugin menu appears.
+, withLua ? false, lua
+}:
+
+stdenv.mkDerivation rec {
+  name = "xournalpp-${version}";
+  version = "1.0.8";
+
+  src = fetchFromGitHub {
+    owner = "xournalpp";
+    repo = "xournalpp";
+    rev = version;
+    sha256 = "01q84xjp9z1krna10gjj562km6i3wdq8cg7paxax1k6bh52ryvf6";
+  };
+
+  # The top-level CMakeLists.txt has a file path that contains `/usr/bin/`,
+  # which `fixCmakeFiles` will helpfully replace with /var/empty. We disable
+  # that step, run it manually, and then un-patch that line prior to
+  # configuring.
+  dontFixCmake = true;
+  preConfigure = ''
+    fixCmakeFiles .
+    sed -e 's|var/empty/local/bin/xopp-recording.sh|usr/local/bin/xopp-recording.sh|g' CMakeLists.txt > CMakeLists.txt.tmp
+    mv CMakeLists.txt.tmp CMakeLists.txt
+  '';
+
+  nativeBuildInputs = [ cmake gettext git makeWrapper pkgconfig ];
```

Comment:
I'm able to build without git.

Reply:
Do you think it's worth keeping? It does print a warning about not being able to find it, but yeah, it builds successfully without it.  Looks like it's primarily used to set a bunch of version information ([1](https://github.com/xournalpp/xournalpp/blob/master/CMakeLists.txt#L20), [2](https://github.com/xournalpp/xournalpp/blob/master/cmake/include/GitRepo.cmake), [3](https://github.com/xournalpp/xournalpp/blob/master/cmake/include/Version.cmake)), which I think is all reproducible, at least.

Reply:
Actually, I answered my own question: `fetchFromGitHub` doesn't appear to keep the `.git` subdirectory, so it's useless anyway :sweat_smile: Removed!